federation/README.md

```diff
@@ -0,0 +1,61 @@
+# Cluster Federation
+
+Kubernetes Cluster Federation enables users to federate multiple
+Kubernetes clusters. Please see the [user guide](http://kubernetes.io/docs/admin/federation/)
+and the [admin guide](http://kubernetes.io/docs/user-guide/federation/federated-services/)
+for more details about setting up and using the Cluster Federation.
+
+# Building Kubernetes Cluster Federation
+
+Please see the [Kubernetes Development Guide](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md)
+for initial setup. Once you have the development environment setup
+as explained in that guide, you also need to install [`jq`](https://stedolan.github.io/jq/download/)
+
+Building cluster federation should be as simple as running:
+
+```shell
+make build do=gen
```

Comment:
There will be neither a do=gen or do=build very soon. See the TODO list below. And also my reply to @mml's comment https://github.com/kubernetes/kubernetes/pull/29515#discussion_r73233835.

For a bit of history, `gen`, 'deploy` and `destroy` were the phrases that kubernetes-anywhere used for the different phases of the lifecycle. Since this work is based on the work in kubernetes-anywhere, it started with the same phrases for ease/simplicity.